msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -97,12 +85,14 @@
             // Is Google App Engine?
             boolean isGAE = System.getProperty("com.google.appengine.runtime.version") != null;
 
-            // For Java6, android and JVM that has no Unsafe class, use Universal MessageBuffer
+            // For Java6, android and JVM that has no Unsafe class, use Universal MessageBuffer (based on ByteBuffer).
+            // Java9 onward doesn't allow to access Cleaner by reflection, so we use Universal MessageBuffer again.
```

Comment:
The performance is the most important factor of msgpack-java. Insteaed of using MessageBufferU for Java9 or Java11, we should have a workaround for the Cleaner issue.

Reply:
I disabled the automatic switch to `MessageBufferU` on Java 9+.